protocol/app/app.go

```diff
@@ -293,6 +305,7 @@ func New(
 	appOpts servertypes.AppOptions,
 	baseAppOptions ...func(*baseapp.BaseApp),
 ) *App {
+	assertAppPreconditions()
```

Comment:
nit: add space afterwards, or include the `appFlags` validation in this function